crates/wasmtime/src/sig_registry.rs

```diff
@@ -80,6 +110,34 @@ impl SignatureRegistry {
         self.wasm2index.get(wasm).cloned()
     }
 
+    /// Builds a lookup table for a module from the possible module's signature
+    /// indices to the shared signature index within this registry.
+    pub fn lookup_table(
+        &self,
+        module: &Module,
+    ) -> PrimaryMap<SignatureIndex, VMSharedSignatureIndex> {
```

Comment:
Rather than returning the `PrimaryMap` and giving up ownership, could we keep the map on `self`, clear it and rebuild it on each call, and reutrn a `&PrimaryMap`? This way we could reuse the underlying allocation.

I suppose -- at that point, though -- we might as well move this out and cache it :sparkles: somewhere :sparkles: like you mention in the OP. I guess finding that somewhere is a bit hard because `SignatureRegistry`s are `Store`-specific, so we can't hang this map off the `Module`.

Reply:
Hm good point! Given what we were discussing on chat though where this may soon move into a `Engine` I think I might leave this as-is for now and we can revisit this later if this gets costly in the future (so far this isn't showing up when benchmarking instantiation for me).